OSPSuite.Presentation.Importer/Presenters/ImporterPresenter.cs

```diff
@@ -0,0 +1,178 @@
+      {
+         _importer = importer;
+         _applicationController = applicationController;
+         _view.AddDataViewingControl(dataViewingPresenter.View);
+         _view.AddColumnMappingControl(columnMappingPresenter.View);
+         _view.AddSourceFileControl(sourceFilePresenter.View);
+         _importer = importer;
+
+         _dataViewingPresenter = dataViewingPresenter;
+         _columnMappingPresenter = columnMappingPresenter;
+         _sourceFilePresenter = sourceFilePresenter;
+
+         _sourceFilePresenter.Title = Captions.Importer.PleaseSelectDataFile;
+         _sourceFilePresenter.Filter = Captions.Importer.ImportFileFilter;
+         _sourceFilePresenter.DirectoryKey = Constants.DirectoryKey.OBSERVED_DATA;
+
+         AddSubPresenters(_dataViewingPresenter, _columnMappingPresenter);
+
+         _sourceFilePresenter.OnSourceFileChanged += onSourceFileChanged;
+         _columnMappingPresenter.OnMissingMapping += onMissingMapping;
+         _columnMappingPresenter.OnMappingCompleted += onCompletedMapping;
+      }
+      public void ImportDataForConfirmation()
+      {
+         OnImportAllSheets.Invoke();
+      }
+      public void ImportDataForConfirmation(string sheetName)
+      {
+
+         OnImportSingleSheet.Invoke(sheetName);
+      }
+
+      public void SetNewFormat(string formatName)
+      {
+         var format = _availableFormats.First(f => f.Name == formatName);
+         SetDataFormat(format, _availableFormats);
+         OnFormatChanged(format.Name);
+      }
+
+      public void GetDataForImport(out string fileName, out IDataFormat format, out IReadOnlyList<ColumnInfo> columnInfos,
+         out IEnumerable<string> namingConventions, out IEnumerable<MetaDataMappingConverter> mappings)
+      {
+         fileName = _dataSourceFile.Path;
+         format = _columnMappingPresenter.GetDataFormat();
+         columnInfos = _columnInfos;
+         namingConventions = _dataImporterSettings.NamingConventions;
+         mappings = _dataSourceFile.Format.Parameters.OfType<MetaDataFormatParameter>().Select(md => new MetaDataMappingConverter()
+         {
+            Id = md.MetaDataId,
+            Index = sheetName => _dataSourceFile.DataSheets[sheetName].RawData.GetColumnDescription(md.ColumnName).Index
+         }).Union
+         (
+            _dataSourceFile.Format.Parameters.OfType<GroupByDataFormatParameter>().Select(md => new MetaDataMappingConverter()
+            {
+               Id = md.ColumnName,
+               Index = sheetName => _dataSourceFile.DataSheets[sheetName].RawData.GetColumnDescription(md.ColumnName).Index
+            })
+         );
+      }
+
+      public IReadOnlyDictionary<string, IDataSheet> GetAllSheets()
+      {
+         return _dataSourceFile.DataSheets;
+      }
+
+      public IDataSheet GetSingleSheet(string sheetName)
+      {
+         return _dataSourceFile.DataSheets[sheetName];
+      }
+
+      //do we really need the arguments in this case???
```

Comment:
IF we are using event, we are using EventHandler and therefor yet. Tis is much easier for testing purposes

Reply:
EventHandler used, thus resolved